### ISSUE TITLE
pyright 1.1.289:  str joining two sets

### ISSUE THREAD

**Issue by FrancescElies** [microsoft/pyright#4470]
**Describe the bug**
The following snippet raises a pyright error, is this designed this way or a bug?
```python
value = " ".join({f"{x}" for x in range(0, 10)} | {f"{x}" for x in range(5, 15)})
print(value)
```
 
**To Reproduce**
```
C:\s\mytest via 🐍 v3.11.1
❯ cat pyright_join_iter_literalstring.py
value = " ".join({f"{x}" for x in range(0, 10)} | {f"{x}" for x in range(5, 15)})
print(value)

C:\s\mytest via 🐍 v3.11.1
❯ python pyright_join_iter_literalstring.py
11 5 9 7 10 12 2 3 13 1 8 0 14 6 4

C:\s\mytest via 🐍 v3.11.1
❯ pyright pyright_join_iter_literalstring.py
No configuration file found.
No pyproject.toml file found.
stubPath C:\s\mytest\typings is not a valid directory.
Assuming Python platform Windows
Searching for source files
Found 1 source file
pyright 1.1.289
C:\s\mytest\pyright_join_iter_literalstring.py
  C:\s\mytest\pyright_join_iter_literalstring.py:1:18 - error: Operator "|" not supported for types "set[str]" and "set[str]"
    Operator "|" not supported for types "set[str]" and "set[str]" when expected type is "Iterable[LiteralString]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 0.612sec

C:\s\mytest via 🐍 v3.11.1 took 3s
❯ pyright --version
pyright 1.1.289
```

**Expected behavior**
For me it was surprising that pyright complained about operator not being supported, code as seen above runs, but maybe I am just missing something?

**Comment by erictraut** [comment]
This is included in pyright 1.1.290, which I just published. It will also be included in a future release of pylance.
